### PR TITLE
Explain more around ASM and other low level concepts

### PR DESCRIPTION
**Pull request by rylev**
Concepts like "architectures" and "word" sizes aren't explain very thoroughly which could lead to some confusion. This is an attempt to explain these concepts a bit better.

Obviously feel free to take as much or little of these changes as you want.

### CODE REVIEW

File: src/tools.md [hunk 1]
```diff
@@ -33,9 +47,42 @@ loop:
...
-
-We can run it like this:
+don't worry. While we could write our entire kernel in assembly, we'll only be
+learning as much assembly as we need to not have to use it any more.
+
+When you write asssembly language you are actually directly manipulating the
+individual registers of the CPU and memory inside of RAM and other hardware
+devices like CD drives or display screens.
+
+CPUs are composed of registers each of which can only hold small amounts of data.
```
Comment (rylev) [inline src/tools.md:57]: This is a good example of a piece of text we can put into an "aside".

File: src/hello-world.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ The `hlt` statement is our first bit of ‘real’ assembly. So far, we had just
 been declaring data. This is actual, executable code. It’s short for ‘halt’.
 In other words, it ends the program.
 
-By giving this line a label, we can call it, like a function. That’s what
+By giving this line a label, we can call it, sort of like a function. That’s what
```
Comment (rylev) [inline src/hello-world.md:24]: Relating labels to functions is helpful now, but might come to bite us later.

File: src/hello-world.md [hunk 3]
```diff
@@ -262,7 +262,7 @@ ENTRY(start)
 ```
 
 This sets the ‘entry point’ for this executable. In our case, we called our
-entry point by the name people use: `start`. Remember? In `boot.s`? Same
```
Comment (rylev) [inline src/hello-world.md:265]: Ah I didn't notice this had already been fixed here: https://github.com/intermezzOS/book/pull/40

**Comment by steveklabnik** [comment]
This is great, thanks so much!

**Pull request merged**